src/main/java/com/epam/lab/payments/UserValidator.java

```diff
@@ -0,0 +1,38 @@
+package com.epam.lab.payments;
+
+import com.epam.lab.payments.domain.UserEntity;
+import com.epam.lab.payments.services.AuthorizationService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+@Component
+@RequiredArgsConstructor
+public class UserValidator implements Validator {
+    private final AuthorizationService authorizationService;
+
+    @Override
+    public boolean supports(Class<?> aClass) {
+        return UserEntity.class.equals(aClass);
+    }
+
+    @Override
+    public void validate(Object o, Errors errors) {
+        UserEntity userEntity = (UserEntity) o;
+
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "username", "NotEmpty");
```

Comment:
Define constants instead of repeating "username" and "password" several times.

Reply:
done